### ISSUE TITLE
Malformed sequence of texture modifiers causes "Floating point exception"

### ISSUE THREAD

**Issue by fluxionary** [minetest/minetest#12393]
##### Minetest version
```
5.5.0, 5.5.1, 5.6.0-dev-ec9f1575121e3b064b919bca7efddfa8b0fc4e65
```

##### OS / Hardware

Operating system: linux
CPU: intel

GPU model: nvidia
OpenGL version: 4.6 (Compatibility Profile) Mesa 22.0.3

##### Summary
A (possibly malformed) sequence of texture modifiers causes `Floating point exception (core dumped)`. A server admin could use this to crash a player's client on a whim.

It seems like there might already be plenty of ways to do that though, e.g. #12214, #11829,  #11822

##### Steps to reproduce
```lua
minetest.register_chatcommand("crash", {
    params = "<victim>",
    description = "Crash the victim's client",
    privs = {server = true},
    func = function(name, victim)
        local player = minetest.get_player_by_name(victim)
        if player then
            player:hud_add({
                hud_elem_type = "image",
                text = "[combine:32x32:1,1=[inventorycube{[combine:16x16{blank.png{blank.png"
            })
            return true, "Player should time out in a moment"
        else
            return false, "Couldn't find the victim"
        end
    end
})
```

**Comment by fluxionary** [comment]
gdb stacktrace:

[stacktrace.txt](https://github.com/minetest/minetest/files/8809273/stacktrace.txt)

**Comment by fluxionary** [comment]
> As shown in your stack trace `[inventorycube{[combine` should work as a minimal reproducer.

I wasn't able to get it to happen w/ anything less than e.g. `[combine:16x16:1,1=[inventorycube{[combine:16x16`. Also, is the sequence really malformed? I actually have a use case for this - I'm trying to pack all 4 smartshop entity textures into a single entity, and some of the individual images might be inventory cubes composed of `[combine:...` faces. Is the upstream https://github.com/minetest/irrlicht/? I might take a stab at trying to fix this.

**Comment by sfan5** [comment]
> Also, is the sequence really malformed?

Yes, colons in texture names need to be escaped for `[combine`.
The fix here would also merely turn the crash into an error message.

**Comment by MCLV-pandorabox** [comment]
> The fix here would also merely turn the crash into an error message btw.

..and that didn't happen. still this issue is closed. :-/

crashed 8 times today before I figured out it was a typo in my ^[resize modifier.

**Comment by sfan5** [comment]
This is supposed to be fixed in 5.6.x, can you show the texture string that crashed?

**Comment by MCLV-pandorabox** [comment]
> This is supposed to be fixed in 5.6.x, can you show the texture string that crashed?

I'm running 5.7.0-dev
and I had a modifier to resize in which I had a silly typo instead of ^[resize:16**x**16  I put in "^[resize:16**:**16"

Would really be great if that would just not work or produce an error instead of client crash :D
